Export/Scripts/legionPassives.lua

```diff
@@ -0,0 +1,195 @@
+for i=1, alternatePassiveSkillDat.rowCount do
+	---@type table<string, boolean|string|number>
+	local datFileRow = {};
+
+	for j=1,#alternatePassiveSkillDat.cols-1 do
+		local key = alternatePassiveSkillDat.spec[j].name
+		datFileRow[key] = alternatePassiveSkillDat:ReadCell(i, j)
+	end
+
+	---@type table<string, boolean|string|number|table>
+	local legionPassiveNode = {};
+
+	-- id
+	legionPassiveNode.id = datFileRow.Id;
+	-- icon
+	legionPassiveNode.icon = datFileRow.DDSIcon;
+	-- is keystone
+	legionPassiveNode.ks = isValueInTable(datFileRow.PassiveType, 4) and true or false
+
+	if legionPassiveNode.ks then ksCount = ksCount + 1 end
+
+	-- is notable
+	legionPassiveNode['not'] = isValueInTable(datFileRow.PassiveType, 3) and true or false
+	-- node name
+	legionPassiveNode.dn = datFileRow.Name;
+	-- is "mastery" ??
+	legionPassiveNode.m = false
+	-- self explanatory
+	legionPassiveNode.isJewelSocket = false
+	legionPassiveNode.isMultipleChoice = false
+	legionPassiveNode.isMultipleChoiceOption = false
+	legionPassiveNode.passivePointsGranted = 0
+	-- something to do with tree starting positions
+	legionPassiveNode.spc = {}
+	-- display text TODO
+	legionPassiveNode.sd = {}
+
+	legionPassiveNode.stats = {}
+
+	for idx,statKey in pairs(datFileRow.StatsKeys) do
+		refRow = statKey._rowIndex
+		statId = stats:ReadCell(refRow, 1)
+		range = datFileRow["Stat"..idx]
+
+		legionPassiveNode.stats[statId] = {
+			["min"] = range[1],
+			["max"] = range[2],
+		}
+	end
+
+	for _, line in ipairs(describeStats(legionPassiveNode.stats)) do
+		table.insert(legionPassiveNode.sd, line)
+	end
+
+	--
+
+	-- TODO node group. everything has a group
+	legionPassiveNode.g = LEGION_PASSIVE_GROUP
+	-- 
+	-- group orbit distance
+	legionPassiveNode.o = legionPassiveNode.ks and 4 or 3
+	legionPassiveNode.oidx = legionPassiveNode.ks and ksCount * 3 or math.floor(math.random() * 1e5)
+	-- attributes granted 
+	legionPassiveNode.sa = 0
+	legionPassiveNode.da = 0
+	legionPassiveNode.ia = 0
+	-- connected nodes
+	legionPassiveNode.out = {}
+	legionPassiveNode["in"] = {}
+	-- legion flag, so pob knows to handle the data in special ways
+	legionPassiveNode.legion = true
+	data.nodes[legionPassiveNode.id] = legionPassiveNode;
+end
+
+data.groups[LEGION_PASSIVE_GROUP] = {
+    ["x"] = -6500,
+    ["y"] = -6500,
+    ["oo"] = {},
+    ["n"] = {}
+}
+
+for k,v in pairs(data.nodes) do
+	table.insert(data.groups[LEGION_PASSIVE_GROUP].n, k)
+end
+
+for i=1, alternatePassiveAdditionsDat.rowCount do
+	---@type table<string, boolean|string|number>
+	local datFileRow = {};
+
+	for j=1,#alternatePassiveAdditionsDat.cols-1 do
+		local key = alternatePassiveAdditionsDat.spec[j].name
+		datFileRow[key] = alternatePassiveAdditionsDat:ReadCell(i, j)
+	end
+
+	---@type table<string, boolean|string|number|table>
+	local legionPassiveAddition = {};
+
+	-- id
+	legionPassiveAddition.id = datFileRow.Id;
+	-- Additions have no name, so we construct one for the UI (also, Lua regex is too limiting :( )
```

Comment:
Lua uses [patterns](http://lua-users.org/wiki/PatternsTutorial) instead of regular expressions. Patterns ~= regex.